app/controllers/spree/admin/payments_controller.rb

```diff
@@ -175,9 +176,11 @@ def authorize_stripe_sca_payment
 
         @payment.authorize!(full_order_path(@payment.order))
 
-        raise Spree::Core::GatewayError, I18n.t('authorization_failure') unless @payment.pending?
+        unless @payment.pending? || @payment.requires_authorization?
+          raise Spree::Core::GatewayError, I18n.t('authorization_failure')
```

Comment:
wouldn't it be worth raising a diferent message in each case? that would at least be helpful while debugging but it could also be useful for the user.